sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -690,7 +690,15 @@ object PushPredicateThroughAggregate extends Rule[LogicalPlan] with PredicateHel
   def apply(plan: LogicalPlan): LogicalPlan = plan transform {
     case filter @ Filter(condition,
         aggregate @ Aggregate(groupingExpressions, aggregateExpressions, grandChild)) =>
-      val (pushDown, stayUp) = splitConjunctivePredicates(condition).partition {
+
+      // Create a map of Alias for grouping keys or literals
+      val aliasMap = AttributeMap(aggregateExpressions.collect {
+        case a: Alias if groupingExpressions.contains(a.child) || a.child.foldable =>
+          (a.toAttribute, a.child)
+      })
+      val newCond = PushPredicateThroughProject.replaceAlias(condition, aliasMap)
```

Comment:
I think this is confusing to read. Not clear why this is calling a utility in PushPredicateThroughProject. Move this into a better named place (similar to PredicateHelper)